wordpress-mobile/WordPress-FluxC-Android
# [Issue/2320] Product variations batch update

**This PR closes #2320 issue. It consists of 2 already reviewed PRs: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2367 and https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2374 which are merged to this feature branch.**

### 🎯 Summary
This PR adds support for product variations batch update, and updates `example` app.

### 🛠 Implementation details
1. Added `/products/<id>/variations/batch endpoint`
2. Implemented POST call to batch update variations `ProductRestClient::batchUpdateVariations`, and introduced `BatchProductVariationsUpdateApiResponse` class wich models backend response.
3. Implemented `WCProductStore::batchUpdateVariations` method responsible for variations batch update on the backend. It updates variations locally after a successful backend response.
4. Created `BatchUpdateVariationsPayload` class consumed by `::batchUpdateVariations` and introduced `BatchUpdateVariationsPayload.Builder` to simplify creating variations updates map required by the REST API.
5. Updated `example` app. Added `WooBatchUpdateVariationsFragment`, accessible from `WooProductsFragment`.

### 📱 Visual changes

<img src="https://user-images.githubusercontent.com/4527432/166924004-34eec8e6-5659-4049-8119-eb1aafcf2935.png" height="512"/> <img src="https://user-images.githubusercontent.com/4527432/166924020-2f4ef1c0-a59f-41d3-990f-aac3235581f9.png" height="512"/>

### 🧪 Testing
* Run WCProductStoreTest. Unit test methods were added here.
* Test with the example app:
    1. Navigate to Batch Update Variations screen.
    2. Enter product id, and ids of variations (comma-separated).
    3. Click Set up product id and variations ids button to validate ids and check if they exist in DB.
    4. Fill in the form below with the data you want to batch update the variations.
    5. Click the button at the bottom.
    
In-app logs inform about the product/variations validation errors and the batch update operation result.
